### PR TITLE
Cache `SubmarinerConfig` only on success

### PR DESCRIPTION
**Pull request by mkolesnik**
Only cache the config on successful operations, otherwise we should
retry cloud prepare/cleanup as long as it wasn't successful.

Resolves https://github.com/stolostron/backlog/issues/24502

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

**Comment by nyechiel** [comment]
@mkolesnik can you please link it to https://github.com/stolostron/backlog/issues/24502? You can edit the first comment and use the "Fixes" keyword. Thanks!

**Comment by skitt** [comment]
See https://github.com/stolostron/submariner-addon/issues/455 for my notes on the test failures (which we’ve seen elsewhere).

File: pkg/hub/submarineragent/controller_test.go [hunk 1]
```diff
@@ -334,6 +334,9 @@ func newTestDriver() *testDriver {
 		providerFactory := cloudFake.NewMockProviderFactory(t.mockCtrl)
 		providerFactory.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(t.cloudProvider, nil).AnyTimes()
 
+		// TODO: Investigate why when tests are a bit slower (e.g run with -v flag) fail due to this being expected.
```
Comment (tpantelis) [inline pkg/hub/submarineragent/controller_test.go:337]: I'm investigating this so we don't need this comment.

Comment (mkolesnik) [reply pkg/hub/submarineragent/controller_test.go:337]: I'd rather merge before 2.6.0 code freeze, I don't mind removing the workaround but UT might fail

Comment (skitt) [inline pkg/hub/submarineragent/controller_test.go:337]:
```suggestion
		// TODO: Investigate why when tests are a bit slower (e.g run with -v flag) fail due to this being unexpected.
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/hub/submarineragent/controller_test.go (hunk 1 above)
Comment (tpantelis) [reply pkg/hub/submarineragent/controller_test.go:337]: The test failure is fixed by #453 (see https://github.com/stolostron/submariner-addon/issues/455#issuecomment-1195605284).  I'd rather not leave a stale comment.

**Review by skitt — APPROVED** [review]

**Comment by nyechiel** [comment]
@skitt @tpantelis Maxim reported https://github.com/stolostron/backlog/issues/24502 with 2.5.2. Can you please cherry-pick this PR into `release-2.5`? Hopefully it's a simple backport that can be handled automatically

**Pull request merged**